varcode/variant.py

```diff
@@ -46,3 +114,240 @@ def short_description(self):
                 chrom, pos + len(alt), pos + len(ref), ref[len(alt):])
         else:
             return "chr%s g.%d %s>%s" % (chrom, pos, ref, alt)
+
+    def transcripts(self):
+        return self.ensembl.transcripts_at_locus(
+            self.contig, self.pos, self.end)
+
+    def coding_transcripts(self):
+        """
+        Protein coding transcripts
+        """
+        return [
+            transcript for transcript in self.transcripts()
+            if is_coding_biotype(transcript.biotype)
+        ]
+
+    def genes(self):
+        return self.ensembl.genes_at_locus(
+            self.contig, self.pos, self.end)
+
+    def coding_genes(self):
+        """
+        Protein coding transcripts
+        """
+        return [
+            gene for gene in self.genes()
+            if is_coding_biotype(gene.biotype)
+        ]
+
+    def annotate(self, raise_on_error=True):
```

Comment:
Sure, I'd be fine with this change. Consider also: `TranscriptModification` instead of `TranscriptEffect`. I don't love either one very much, but the "modification" language at least resolves the effect-on-transcript vs. effect-of-transcript ambiguity